src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -152,11 +152,27 @@ private void EnsureDefaultCert()
                 var logger = ApplicationServices.GetRequiredService<ILogger<KestrelServer>>();
                 try
                 {
-                    DefaultCertificate = CertificateManager.ListCertificates(CertificatePurpose.HTTPS, StoreName.My, StoreLocation.CurrentUser, isValid: true)
+                    DefaultCertificate = CertificateManager.Instance.ListCertificates(StoreName.My, StoreLocation.CurrentUser, isValid: true)
                         .FirstOrDefault();
 
                     if (DefaultCertificate != null)
                     {
+                        var status = CertificateManager.Instance.CheckCertificateState(DefaultCertificate, interactive: false);
+                        if (!status.Result)
+                        {
+                            // Display a warning indicating to the user that a prompt might appear and provide instructions on what to do in that
+                            // case. The underlying implementation of this check is specific to Mac OS and is handled within CheckCertificateState.
```

Comment:
```suggestion
                            // case. The underlying implementation of this check is specific to Mac OS and is handled within CheckCertificateState.
                            // Kestrel must NEVER cause a UI prompt on a production system. We only attempt this here because Mac OS is not supported
                            // in production.
```